src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx

```diff
@@ -0,0 +1,124 @@
+import TALENTS from 'common/TALENTS/paladin';
+import { SpellLink, TooltipElement } from 'interface';
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import Events, {
+  AnyEvent,
+  CastEvent,
+  EventType,
+  GetRelatedEvents,
+  ResourceChangeEvent,
+} from 'parser/core/Events';
+import BoringValueText from 'parser/ui/BoringValueText';
+import ItemManaGained from 'parser/ui/ItemManaGained';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { DAYBREAK_HEALING, DAYBREAK_MANA } from '../../normalizers/CastLinkNormalizer';
+import { formatNumber } from 'common/format';
+
+class Daybreak extends Analyzer {
+  castEvents: CastEvent[] = [];
+
+  manaGained: number = 0;
+  manaWasted: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.DAYBREAK_TALENT);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(
+      Events.cast.spell(TALENTS.DAYBREAK_TALENT).by(SELECTED_PLAYER),
+      this.onCast,
+    );
+  }
+
+  onCast(event: CastEvent) {
+    this.castEvents.push(event);
+
+    const manaEvents = GetRelatedEvents(event, DAYBREAK_MANA);
+    manaEvents.forEach((e: AnyEvent) => {
```

Comment:
IDK if its possible but is there any way we can not do `AnyEvent`... maybe `.filter()` before hand?
Maybe an ugly cast on the whole array?

Reply:
harder to change the type here, but easy to change the next line to `if (e.type === EventTypes.ResourceChange)` which i think is nicer than the explicit cast. decent resolution?

Reply:
That works but also not the end of the world if you just wanna send it